js/everything.js

```diff
@@ -225,7 +230,44 @@ var ZenPlayer = {
                 updateTweetMessage();
             });
 
-            plyrPlayer.addEventListener("timeupdate", function() {
+            plyrPlayer.addEventListener("ended", function () {
+                videoMetadata = getParsedvideoMetadata();
+                var index = -1;
+                if (videoMetadata) {
+                    var hasDuplicates = function (arr) {
+                        var len = arr.length;
+                        for (var i = 0; i < len; i++) {
+                            if (arr[i].id === currentVideoID) {
+                                return i;
+                            }
+                        }
+                        return -1;
+                    };
+                    index = hasDuplicates(videoMetadata.items);
```

Comment:
Looks like this function is actually doing 2 things, and should probably be renamed:

1. Return -1 if all items are unique
2. Return the index of the duplicate